settings.py

```diff
@@ -44,27 +58,44 @@
 LOGGING['root']['level'] = env('DJANGO_LOG_LEVEL', 'INFO')
 
 AWS_STORAGE_BUCKET_NAME = env('AWS_STORAGE_BUCKET_NAME', '')
+AWS_BUCKET_DOMAIN = env('AWS_BUCKET_DOMAIN', AWS_STORAGE_BUCKET_NAME + '.s3.amazonaws.com')
 CDN_DOMAIN_NAME = env('CDN_DOMAIN_NAME', '')
+AWS_ACCESS_KEY_ID = env('AWS_ACCESS_KEY_ID', '')
+AWS_SECRET_ACCESS_KEY = env('AWS_SECRET_ACCESS_KEY', '')
+AWS_S3_REGION_NAME = env('AWS_S3_REGION_NAME', '')
+AWS_DEFAULT_ACL = env('AWS_DEFAULT_ACL', '')
+AWS_LOCATION = env('AWS_LOCATION', '')
+AWS_STATIC = env('AWS_STATIC', False)
+AWS_MEDIA = env('AWS_MEDIA', False)
 
 if AWS_STORAGE_BUCKET_NAME:
     # Tell django-storages that when coming up with the URL for an item in S3 storage, keep
     # it simple - just use this domain plus the path. (If this isn't set, things get complicated).
     # This controls how the `static` template tag from `staticfiles` gets expanded, if you're using it.
     # We also use it in the next setting.
     if CDN_DOMAIN_NAME:
-        AWS_S3_CUSTOM_DOMAIN = CDN_DOMAIN_NAME
+        AWS_S3_DOMAIN = CDN_DOMAIN_NAME
     else:
-        AWS_S3_CUSTOM_DOMAIN = '%s.s3.amazonaws.com' % AWS_STORAGE_BUCKET_NAME
+        AWS_S3_DOMAIN = '%s.s3.amazonaws.com' % AWS_STORAGE_BUCKET_NAME
+
+    if AWS_STATIC:
+        # This is used by the `static` template tag from `static`, if you're using that. Or if anything else
+        # refers directly to STATIC_URL. So it's safest to always set it.
+        STATIC_URL = "https://%s/" % AWS_S3_DOMAIN
+
+        # Tell the staticfiles app to use S3Boto storage when writing the collected static files (when
+        # you run `collectstatic`).
+        STATICFILES_STORAGE = 'storages.backends.s3boto3.S3Boto3Storage'
+
+        COMPRESS_STORAGE = STATICFILES_STORAGE
+
+    if AWS_MEDIA:
+        MEDIAFILES_LOCATION = 'media'
+        MEDIA_URL = "https://%s/%s/" % (AWS_S3_DOMAIN, MEDIAFILES_LOCATION)
```

Comment:
I'd rather use the bucket URL directly for uploaded media since we can't control the file names people use, and hence may have issues with invalidation. What do you think?

Reply:
@tobiasmcnulty not sure I understand, can you elaborate please?

Reply:
I think it's better to use the bucket domain (not the CDN domain, if configured).